core/consensus/grandpa/impl/grandpa_impl.cpp

```diff
@@ -119,19 +122,22 @@ namespace kagome::consensus::grandpa {
 
     // allow app state manager to prepare, start and stop grandpa consensus
     // pipeline
-    app_state_manager.takeControl(*this);
+    app_state_manager_->takeControl(*grandpa_pool_handler_);
+    app_state_manager_->takeControl(*this);
   }
 
-  bool GrandpaImpl::start() {
-    if (!grandpa_pool_handler_->isInCurrentThread()) {
-      grandpa_pool_handler_->execute([wptr{weak_from_this()}] {
-        if (auto self = wptr.lock()) {
-          self->start();
+  void GrandpaImpl::start() {
+    grandpa_pool_handler_->postAlways([wptr{weak_from_this()}] {
+      if (auto self = wptr.lock()) {
+        if (not self->tryStart()) {
+          SL_ERROR(self->logger_, "start failed");
```

Comment:
Please, make log message more clean. 
And, makes sense to use critical-level of log